oracle/types/oracle.go

```diff
@@ -24,116 +23,121 @@ type (
 	// PriceProviderFactory is a type alias for the price provider factory. This
 	// specifically only returns price providers that implement the provider interface
 	// and the additional base provider methods.
-	PriceProviderFactory = factory.BaseProviderFactory[mmtypes.Ticker, *big.Int]
+	PriceProviderFactory = factory.BaseProviderFactory[ProviderTicker, *big.Float]
 
 	// PriceProviderFactory is a type alias for the price provider factory. This
 	// specifically only returns price providers that implement the provider interface.
-	PriceProviderFactoryI = factory.ProviderFactory[mmtypes.Ticker, *big.Int]
+	PriceProviderFactoryI = factory.ProviderFactory[ProviderTicker, *big.Float]
 
 	// PriceProvider is a type alias for the base price provider. This specifically
 	// implements the provider interface for the price provider along with the
 	// additional base provider methods.
-	PriceProvider = base.Provider[mmtypes.Ticker, *big.Int]
+	PriceProvider = base.Provider[ProviderTicker, *big.Float]
 
 	// PriceProviderI is a type alias for the price provider. This specifically
 	// implements the provider interface for the price provider.
-	PriceProviderI = providertypes.Provider[mmtypes.Ticker, *big.Int]
+	PriceProviderI = providertypes.Provider[ProviderTicker, *big.Float]
```

Comment:
These feel very similar and are confusing. Can we consolidate into a single alias/interface for each of these?